heads.go

```diff
@@ -107,12 +107,15 @@ func (hh *heads) Replace(h, c cid.Cid, height uint64) error {
 			return err
 		}
 	}
-	return nil
+	return hh.store.Sync(hh.namespace)
```

Comment:
I am not sure I follow. What final commit?

Reply:
When the user explicitly calls `Sync`. That is, I'm wondering why we're syncing before then.

Reply:
ok, my thinking here was that, since the Heads are internal accounting unrelated to any key or prefix the user Syncs() they should be synced internally, specially since Heads might change without the user doing anything. But so can every other key, so I guess you are right. I'll change it.